src/storm-pomdp/parser/PomdpSolveParser.cpp

```diff
@@ -0,0 +1,422 @@
+#include "PomdpSolveParser.h"
+#include "storm/adapters/RationalNumberAdapter.h"
+#include "storm/api/builder.h"
+#include "Utility.h"
+#include <fstream>
+#include <sstream>
+#include <string>
+#include <vector>
+#include <map>
+#include <unordered_set>
+#include <iostream>
+
+namespace storm {
+namespace pomdp {
+namespace parser {
+
+struct POMDP {
+    std::vector<std::string> states;
+    std::vector<std::string> actions;
+    std::vector<std::string> observations;
+    std::vector<double> start_probabilities;
+    std::map<std::string, std::map<std::string, double>> transitions;
+    std::map<std::string, std::map<std::string, double>> observations_prob;
+    std::map<std::string, double> rewards;
+    std::vector<std::string> newStates; // Stores unique new states with observation incorporated
+    std::map<std::string, std::map<std::string, double>> newTransitions; // Modified transitions with observation-based states
+
+    std::map<std::string, std::map<std::string, double>> createNewTransitions() {
+        std::unordered_set<std::string> uniqueStates;
+
+        for (const auto& action_entry : transitions) {
```

Comment:
Style: we typically use ``TYPE const`` instead of ``const TYPE``; not wrong, just inconsistent with the rest of the code